Widgets_CDAT/DictWidget/KeyValueEditor.py

```diff
@@ -0,0 +1,274 @@
+from PySide.QtCore import *
+from PySide.QtGui import *
+from functools import partial
+import pdb
+
+class InputChecker(QValidator):
+
+    inputInvalid = Signal()
+    correctInput = Signal()
+
+    def __init__(self, index):
+        super(InputChecker, self).__init__()
+        self.texts = []
```

Comment:
Instead of manually updating these, how about you pass in a reference to the DictEditor object, and just fetch them from there every time you do `validate`? Then you don't need to worry about holding onto references to these.